route/build/src/process_param.f90

```diff
@@ -56,23 +56,33 @@ SUBROUTINE basinUH(dt, fshape, tscale, IERR, MESSAGE)
   ! ---------------------------------------------------------------------------------------
   ! initialize error control
   ierr=0; message='basinUH/'
-  ! use a Gamma distribution with shape parameter, fshape = 2.5, and time parameter, tscale, input
+  ! use a Gamma distribution with shape parameter, fshape, and time parameter, tscale, input
   alamb = fshape/tscale                  ! scale parameter
   ! find the desired number of future time steps
-  ntdh_min = 1._dp
-  ntdh_max = 1000._dp
-  ntdh_try = 0.5_dp*(ntdh_min + ntdh_max)
-  do itry=1,maxtry
-   x_value = alamb*dt*ntdh_try
-   cumprob = gammp(fshape, x_value)
-   !print*, tscale, ntdh_try, cumprob
-   if(cumprob < 0.99_dp)  ntdh_min = ntdh_try
-   if(cumprob > 0.999_dp) ntdh_max = ntdh_try
-   if(cumprob > 0.99_dp .and. cumprob < 0.999_dp) exit
-   ntdh_try = 0.5_dp*(ntdh_min + ntdh_max)
-   if(itry==maxtry)then; ierr=20; message=trim(message)//'cannot identify the maximum number of bins for the tdh'; return; endif
-  end do
+  ! in case check if the cummulative Gamma distribution is 1 for given model time step, tscale and fsahpe.
+  ! Modified by Shervan Gharari on 23th July 2019
```

Comment:
You should not leave names embedded in the code. This gets very messy very quickly. GitHub tracks who did what when.

Reply:
right! got it...